src/nevermined/api/NVMBaseApi.ts

```diff
@@ -0,0 +1,296 @@
+
+            let encryptedFiles
+            if (!['workflow'].includes(assetAttributes.metadata.main.type)) {
+                const encryptedFilesResponse = await this.nevermined.services.node.encrypt(
+                    ddo.id,
+                    JSON.stringify(assetAttributes.metadata.main.files),
+                    new String(assetAttributes.encryptionMethod)
+                )
+                encryptedFiles = JSON.parse(encryptedFilesResponse)['hash']
+            }
+
+            let serviceEndpoint = this.nevermined.services.metadata.getServiceEndpoint(
+                DID.parse(ddo.id)
+            )
+
+            await ddo.updateService(this.nevermined, {
+                type: 'metadata',
+                index: 0,
+                serviceEndpoint,
+                attributes: {
+                    // Default values
+                    curation: {
+                        rating: 0,
+                        numVotes: 0
+                    },
+                    // Overwrites defaults
+                    ...assetAttributes.metadata,
+                    encryptedFiles,
+                    // Cleaning not needed information
+                    main: {
+                        ...assetAttributes.metadata.main,
+                        files: assetAttributes.metadata.main.files?.map((file, index) => ({
+                            ...file,
+                            index,
+                            url: undefined
+                        }))
+                    } as any
+                }
+            } as Service)
+
+            this.logger.log('Files encrypted')
+            observer.next(CreateProgressStep.FilesEncrypted)
+
+            const checksum = ddo.getProofChecksum()
+
+            const ddoVersion: NvmConfigVersions = {
+                id: 0,
+                updated: ddo.created,
+                checksum,
+                immutableUrl: ''
+            }
+            ddo._nvm.versions.push(ddoVersion)
+
+            if (publishMetadata != PublishMetadata.OnlyMetadataAPI) {
+                observer.next(CreateProgressStep.DdoStoredImmutable)
+                try {
+                    ({url: ddoVersion.immutableUrl, backend: ddoVersion.immutableBackend } = 
+                        await this.nevermined.services.node.publishImmutableContent(ddo, publishMetadata))
+                } catch (error) {
+                    this.logger.log(`Unable to publish immutable content`)
+                }
+                
+            }
+
+            if (ddoVersion.immutableBackend)
+                ddo._nvm.versions[0] = ddoVersion
+
+            observer.next(CreateProgressStep.RegisteringDid)
+
+            // On-chain asset registration
+            if (nftAttributes) {
+                this.logger.log('Registering Mintable Asset', ddo.id)
+                
+                const nftAttributesWithoutRoyalties = { ...nftAttributes}
+                nftAttributesWithoutRoyalties.royaltyAttributes = undefined
```

Comment:
I don't fully understand this. shouldn't `registerMintableDID` not care about royalties if they are being set in a separate call?

Reply:
contracts expose `registerMintableDID721` function that allow to specify some basic royalties, and we use `DIDRegistry.registerMintableDID721` method that integrates that function. Because we want to support in the `create` process a more sophisticated royalties configuration we skip that configuration there to do it later but still re-using the `registerMintableDID721`